pkg/controller/sparkapplication/controller.go

```diff
@@ -447,8 +440,21 @@ func (c *Controller) syncSparkApplication(key string) error {
 	switch appToUpdate.Status.AppState.State {
 	case v1alpha1.NewState:
 		c.recordSparkApplicationEvent(appToUpdate)
-		appToUpdate.Status.SubmissionAttempts = 0
 		appToUpdate = c.submitSparkApplication(appToUpdate)
+	case v1alpha1.PendingSubmissionState:
+		succeeded, err := c.subJobManager.hasJobSucceeded(appToUpdate)
+		if err != nil {
+			return err
+		}
+		if succeeded != nil {
+			if *succeeded {
+				c.createSparkUIResources(appToUpdate)
+				appToUpdate.Status.AppState.State = v1alpha1.SubmittedState
+				appToUpdate.Status.ExecutionAttempts += 1
+			} else {
+				appToUpdate.Status.AppState.State = v1alpha1.FailedSubmissionState
```

Comment:
Looks like FailedSubmissionState is a terminal state now as we end up in this only after exhausting all the retries. This is a non-backward compatible change since till now only Failed/Completed were terminal states. Does it make sense to deprecate FailedSubmissionState and move this to Failed here to be backward compatible ?

Reply:
Makes sense to me.